li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -238,6 +242,15 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
       Integer partition = producerRecord.partition();
       Future<RecordMetadata> future = null;
       UUID messageId = _uuidFactory.getUuid(producerRecord);
+
+      Headers headers = producerRecord.headers();
+      if (headers == null) {
+        headers = new RecordHeaders();
+      }
+      // Remove any header that maybe using the key for audit event timestamp or large message.
```

Comment:
"_" prefixed headers are meant to be used internally (yet to update the documentation). That said, we don't really want end-users to set these values. They can communicate the record timestamp via ProducerRecord#timestmap() though. // Sticking with the earlier auditing behavior.

Reply:
> Also, should the actual timestamp always be the System instead of ProducerRecord.timestamp()?

This is what I had in my earlier implementation but changed it so that it is now equivalent to the Auditor behavior.